tests/Integration/OtherTest.php

```diff
@@ -213,6 +213,96 @@ public function test_checking_site_id_is_set_or_not(): void {
 		self::assertFalse( self::$parsely->site_id_is_missing() );
 	}
 
+	/**
+	 * Verifies that managed credentials get set as expected.
+	 *
+	 * @since 3.9.0
+	 *
+	 * @covers \Parsely\Parsely::are_credentials_managed
+	 * @covers \Parsely\Parsely::inject_managed_credentials
+	 * @uses \Parsely\Parsely::__construct
+	 * @uses \Parsely\Parsely::get_default_options
+	 */
+	public function test_managed_credentials_get_set_as_expected(): void {
+		// Valid and invalid credential sets to be tested.
+		$credential_sets = array(
```

Comment:
Here I think we should provide array as following:

```php
array(
    array(
        'input' => array(), // having credentials
        'expected' => array(), // having expected credentials
     ),
      array(
        'input' => array(), // having credentials
        'expected' => array(), // having expected credentials
     )
)
```

More changes will be that we don't need to invoke the credentials function ourselves since we can use the existing `get_options` function.

The main benefits of this approach are

1. Easier because we just have to compare the returned values with `expected` array and we can avoid this function

```php
if ( is_array( $credentials ) && 0 < count( $credentials ) ) {
				foreach ( $credentials as $key => $value ) {
					if ( 'site_id' === $key ) {
						self::assertSame( $actual_options['apikey'], $value );
					} elseif ( 'api_secret' === $key ) {
						self::assertSame( $actual_options['api_secret'], $value );
					} elseif ( 'metadata_secret' === $key ) {
						self::assertSame( $actual_options['metadata_secret'], $value );
					} else {
						// Invalid keys should not be added to the options array.
						self::assertFalse( isset( $actual_options[ $key ] ) );
					}
				}
			} else {
				// Credentials are of wrong data type or an empty array.
				self::assertSame( $default_options, $actual_options );
			}
```

2. Easier to add new cases because we don't have that function

3. We can use credentials type (if defined above) to make it easy to understand the code

Reply:
Updated but left some asserts for options, since the functions that return the options and the managed credentials are different (getting the correct managed credentials does not mean that they get correctly merged into the options array).